homeassistant/components/immich/coordinator.py

```diff
@@ -48,6 +49,10 @@ def __init__(
         """Initialize the data update coordinator."""
         self.api = api
         self.is_admin = is_admin
+        self.configuration_url = (
+            f"{'https' if entry.data[CONF_SSL] else 'http'}://"
+            f"{entry.data[CONF_HOST]}:{entry.data[CONF_PORT]}"
+        )
```

Comment:
Would using YARL make this simpler?

Reply:
not sure, because we have to define the protocol even in yarl 🤔